src/hotspot/share/runtime/thread.cpp

```diff
@@ -3022,11 +3012,13 @@ void Threads::initialize_java_lang_classes(JavaThread* main_thread, TRAPS) {
   call_initPhase1(CHECK);
 
   // get the Java runtime name, version, and vendor info after java.lang.System is initialized
-  JDK_Version::set_java_version(get_java_version(THREAD));
-  JDK_Version::set_runtime_name(get_java_runtime_name(THREAD));
-  JDK_Version::set_runtime_version(get_java_runtime_version(THREAD));
-  JDK_Version::set_runtime_vendor_version(get_java_runtime_vendor_version(THREAD));
-  JDK_Version::set_runtime_vendor_vm_bug_url(get_java_runtime_vendor_vm_bug_url(THREAD));
+  InstanceKlass* ik = SystemDictionary::find_instance_klass(vmSymbols::java_lang_VersionProps(),
+                                                            Handle(), Handle());
+  JDK_Version::set_java_version(get_java_version(ik));
```

Comment:
The various get_java_xxx() functions all seem to do the same thing. I am wondering if they can be combined into a single utility function, so that you can do something like:

```
JDK_Version::set_runtime_name(get_version_info(ik, vmSymbols::java_version_name(),
                              java_version, sizeof(java_version)));
```

Reply:
I think the get_java_* set of functions could be streamlined so that you pass in the symbol for the field you need. Also perhaps VersionProps could be a well-known class so we don't have to look it up. But this seems a future RFE. Even the current changes seem a little out-of-scope for this change.

Reply:
yes, this might be a useful change in a follow up RFE. What you have is good.  You were going to change the lines anyway.

Reply:
Hi Ioi, I'd like to do this cleanup in a separate RFE.  Is that okay?

Reply:
Hi Harold, I think it's OK to do the cleanups in different RFEs.